### ISSUE TITLE
Add-on ddoes not work on NVDA 2024.3

### ISSUE THREAD

**Issue by CyrilleB79** [mahmoodhozhabri/VisionAssistantPro#49]
According to [this thread](https://groups.io/g/nvda-it/topic/vision_assistant_inattivo/116790677), the add-on does not start up with NVDA 2024.3 although it claims to be compatible with it.

Therror is as follows:

```
ERROR - globalPluginHandler.listPlugins (20:10:51.398) - MainThread (20436):
Error importing global plugin visionAssistant
Traceback (most recent call last):
  File "globalPluginHandler.pyc", line 23, in listPlugins
  File "importlib\__init__.pyc", line 126, in import_module
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "C:\Users\pierp\AppData\Roaming\nvda\addons\VisionAssistant\globalPlugins\visionAssistant\__init__.py", line 25, in <module>
    import markdown
ModuleNotFoundError: No module named 'markdown'
```

The markdown module has not always been in NVDA.

Solutions:
* either you reduce compatibility range
* or you include markdown module in your add-on to continue supporting earlier releases of NVDA
* or you disable features requiring markdown for older releases of NVDA which do not embed it.

**Comment by mahmoodhozhabri** [comment]
@CyrilleB79 Hello. Thank you for letting me know. Since which version has this module existed? I will reduce the compatibility.

**Comment by mahmoodhozhabri** [comment]
Thank you for the advice.

I have implemented the fix using a context manager to ensure `sys.path` is cleaned immediately after the import. This improvement will be included in the upcoming release (v3.0.2).
